tensorflow/python/framework/python_op_gen.cc

```diff
@@ -367,6 +432,108 @@ string GenEagerPythonOp::Code() {
   return prelude_ + result_;
 }
 
+std::unordered_map<string, string> GenEagerPythonOp::GetTypeAnnotationMap() {
+  std::unordered_map<string, string> type_annotations;
```

Comment:
I recommend using two maps to avoid mixing attrs with inputs/outputs: for example, attr_types and input_output_types.

Reply:
Resolving this since we spoke offline about how the op's parameters include both inputs and attrs.